crates/indexer/src/geyser/programs/magic_eden_haus.rs

```diff
@@ -0,0 +1,149 @@
+    buyer_price: u64,
+    token_size: u64,
+    expiry: u64,
+}
+
+pub(crate) async fn process_sale(
+    client: &Client,
+    mut data: &[u8],
+    accounts: &[Pubkey],
+    slot: u64,
+) -> Result<()> {
+    let params = MEInstructionData::deserialize(&mut data)
+        .context("failed to deserialize ME instructions")?;
+
+    let accts: Vec<String> = accounts.iter().map(ToString::to_string).collect();
+
+    let row = SellInstruction {
+        wallet: Owned(accts[0].clone()),
+        token_account: Owned(accts[5].clone()),
+        metadata: Owned(accts[4].clone()),
+        authority: Owned(accts[6].clone()),
+        auction_house: Owned(accts[7].clone()),
+        // TODO: make this optional
+        auction_house_fee_account: Borrowed("magice_eden_haus"),
+        // TODO: make this optional
+        seller_trade_state: Borrowed("magice_eden_haus"),
+        // TODO: make this optional
+        free_seller_trader_state: Borrowed("magice_eden_haus"),
+        // TODO: make this optional
+        program_as_signer: Borrowed("magice_eden_haus"),
+        trade_state_bump: params.trade_state_bump.try_into()?,
+        free_trade_state_bump: 250,
+        program_as_signer_bump: 250,
+        buyer_price: params.buyer_price.try_into()?,
+        token_size: params.token_size.try_into()?,
+        created_at: Utc::now().naive_utc(),
+        slot: slot.try_into()?,
+    };
+
+    upsert_into_listings_table(client, row.clone())
+        .await
+        .context("failed to insert listing!")?;
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(sell_instructions::table)
+                .values(&row)
+                .execute(db)
+        })
+        .await
+        .context("failed to insert sell instruction ")?;
+
+    Ok(())
+}
+
+pub(crate) async fn process_buy(
+    client: &Client,
+    mut data: &[u8],
+    accounts: &[Pubkey],
+    slot: u64,
+) -> Result<()> {
+    let params = MEInstructionData::deserialize(&mut data)
+        .context("failed to deserialize ME  instructions")?;
+
+    // ME2 accounts
+    if accounts.len() != 12 {
+        debug!("invalid accounts for BuyInstruction");
+        return Ok(());
+    }
+
+    let accts: Vec<String> = accounts.iter().map(ToString::to_string).collect();
+
+    let row = BuyInstruction {
+        wallet: Owned(accts[0].clone()),
+        payment_account: Owned(accts[0].clone()),
+        transfer_authority: Owned(accts[1].clone()),
+        treasury_mint: Borrowed("So11111111111111111111111111111111111111112"),
+        token_account: Owned(accts[2].clone()),
+        metadata: Owned(accts[3].clone()),
+        escrow_payment_account: Owned(accts[4].clone()),
+        authority: Owned(accts[5].clone()),
+        auction_house: Owned(accts[6].clone()),
+        // TODO: make this optional?
+        auction_house_fee_account: Borrowed("magice_eden_haus"),
+        // TODO: make this optional
+        buyer_trade_state: Borrowed("magice_eden_haus"), //
```

Comment:
offers has buyer_trade_state which is not present in ME Buy ins so maybe we should make it optional @kespinola